Source/Scene/ModelUtility.js

```diff
@@ -167,9 +167,18 @@ define([
         }
     };
 
+    function regexIndexOf(text, regex, i) {
+        var indexInSuffix = text.slice(i).search(regex);
+        return indexInSuffix < 0 ? indexInSuffix : indexInSuffix + i;
+    }
+
     function replaceAllButFirstInString(string, find, replace) {
-        var index = string.indexOf(find);
-        return string.replace(new RegExp(find, 'g'), function(match, offset) {
+        // Limit search to strings that are not a subset of other tokens.
+        find += '(?![a-zA-Z_$])';
+        find = new RegExp(find, 'g');
+
+        var index = regexIndexOf(string, find, 0);
```

Comment:
Could this line just be replaced with `var index = string.search(find);` and `regexIndexOf` be removed?